### PR TITLE
Adding couple of missing test cases in policy service

### PR DESCRIPTION
**Pull request by Jasintha**

### CODE REVIEW

File: components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/main/java/org/wso2/carbon/policy/mgt/core/PolicyManagerServiceImpl.java [hunk 1]
```diff
@@ -149,8 +149,10 @@ public Policy getEffectivePolicy(DeviceIdentifier deviceIdentifier) throws Polic
                         deviceIdentifier.getId() + " - " + deviceIdentifier.getType());
             }
         } catch (PolicyEvaluationException e) {
-            String msg = "Error occurred while getting the effective features from the PEP service " +
-                    deviceIdentifier.getId() + " - " + deviceIdentifier.getType();
+
```
Comment (sinthuja) [inline components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/main/java/org/wso2/carbon/policy/mgt/core/PolicyManagerServiceImpl.java]: Can you remove these blank lines?

File: components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/test/java/org/wso2/carbon/policy/mgt/core/PolicyManagerServiceImplTest.java [hunk 2]
```diff
@@ -212,6 +218,7 @@ public void addPolicy() throws DeviceManagementException, GroupManagementExcepti
         policy1.setTenantId(PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId());
         policy1 = policyManagerService.addPolicy(policy1);
 
+
```
Comment (sinthuja) [inline components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/test/java/org/wso2/carbon/policy/mgt/core/PolicyManagerServiceImplTest.java]: Can you remove these unnecessary blank lines?

File: components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/test/java/org/wso2/carbon/policy/mgt/core/PolicyManagerServiceImplTest.java [hunk 3]
```diff
@@ -326,9 +338,38 @@ public void inactivatePolicy() throws PolicyManagementException {
...
+        }
+
+    }
+
+    @Test(dependsOnMethods = "applyPolicy")
+    public void getDeviceCompliance() throws Exception{
+        NonComplianceData deviceCompliance = policyManagerService.getDeviceCompliance(new DeviceIdentifier(DEVICE1, DEVICE_TYPE_A));
+        Assert.assertNotNull(deviceCompliance);
+    }
+
```
Comment (sinthuja) [inline components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/test/java/org/wso2/carbon/policy/mgt/core/PolicyManagerServiceImplTest.java]: Can you remove these unnecessary blank lines?

File: components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/test/java/org/wso2/carbon/policy/mgt/core/PolicyManagerServiceImplTest.java [hunk 4]
```diff
@@ -326,9 +338,38 @@ public void inactivatePolicy() throws PolicyManagementException {
...
+    @Test(dependsOnMethods = "applyPolicy")
+    public void getEffectiveFeatures( ) throws Exception {
+        List<ProfileFeature> effectiveFeatures = policyManagerService.
+                getEffectiveFeatures(new DeviceIdentifier(DEVICE1, DEVICE_TYPE_A));
+        Assert.assertNotNull(effectiveFeatures);
+        Assert.assertEquals(POLICY1_FEATURE1_CODE,effectiveFeatures.get(0).getFeatureCode());
+        try{
+              policyManagerService.getEffectiveFeatures(new DeviceIdentifier(DEVICE_2, DEVICE_TYPE_B));
+        }catch(FeatureManagementException e){
+           Assert.assertTrue(e.getCause()instanceof PolicyEvaluationException);
```
Comment (sinthuja) [inline components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/test/java/org/wso2/carbon/policy/mgt/core/PolicyManagerServiceImplTest.java]: Please throw back the exception if the exception cause is not an instance of the PolicyEvaluationException. We need to throw back all exceptions, rather trying to assert the exceptions there, else during the test failures due to unexpected exceptions, we'll not get the full trace.

**Review by sinthuja — DISMISSED** [review]
Please fix the PR based on the comments.

**Comment by Jasintha** [comment]
This PR is WIP one (Rasika want this changes)that's why you see some formatting issue ,anyway thanks for pointing and mistakenly I haven't to add the WIP tag :(

**Review by ruwany — APPROVED** [review]

**Pull request merged**